contribute/design/proposal-cstor-orchestration.md

```diff
@@ -0,0 +1,311 @@
+- Knowledge about Kubernetes [CRDs](https://kubernetes.io/docs/concepts/api-extension/custom-resources/)
+- Knowledge about Kubernetes [Resource limits and requests for CPU and Memory](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#resource-requests-and-limits-of-pod-and-container)
+- Knowledge about Kubernetes concepts like Custom Controllers, initializers and reconcilation loop that wait on objects to move from actual to desired state.
+- Familiar with Kubernetes and OpenEBS Storage Concepts:
+  * PersistentVolume(PV) and PersistentVolumeClaim(PVC) are standard kubernetes terms used to associate volumes to a given workload. A PVC will be watched by a dynamic provisioner, which will inturn spin-up OpenEBS Volume Containers and create a PV (iSCSI) to be used by the workload.
+  * PersistentDisks(PDs) and PersistentDiskClaims(PDCs) are used to represent and identify the storage (a disk) attached to a Kubernetes Node. PersistentDiskClaims are created by the User and maya-nodebot will use that information to discover disks matching the criteria mentioned in PDC to create the PDs. It is also possible that the User will directly create PD representing a storage attached to a node. Each PD will be represented by a cluster wide unique identifier like -- rs://host:nodea/device:/dev/disk-by-id/avcd-121312-hfjgdf. (TODO - Provide the link to the design documented here)
+  * StoragePool(SP) and StoragePoolClaim(SPC) are used to respresent and identify a pool of storage that is created using one more PDs. Each StoragPool can be used to then serve one or more OpenEBS Volume Replicas. The StoragePools can be of different types depending on how the underlying disks/storage are aggregated and how the pool is exposed. Till 0.6, OpenEBS supported storage pools of type (filesystem), which where basically either a hostDirectory or ext4 mount of a single block disk. From 0.7, OpenEBS will support creation of storage pools that expose blocks. These block type pools are implemented using cStor containers. 
+
+
+## Design
+
+The design is split into two aspects - creation of storage pools using cStor Engine and creating OpenEBS Volumes using the cStor engine. This implementation makes use of CRDs to storage the disk, pool and volume related information. As part of the openebs installation, the followind CRDs are loaded:
+- PeristentDisk and PersistentDiskClaim
+- StoragePool and StoragePoolClaim
+- cStorPool
+- cStorReplica
+(TODO: @ganesh) Fill in the YAMLs for the above CRDs. 
+
+### Creating cStor StoragePools:
+
+   * Admin will create StoragePoolClaim(SPC) with type=block and the SPC will contains information like:
+     ```
+     apiVersion: openebs.io/v1alpha1
+     kind: StoragePoolClaim
+     metadata:
+       name: pool1
+       type: block
```

Comment:
Type should be in the specs than metadata

Reply:
changed